internal/eval/store_eval.go

```diff
@@ -1290,3 +1290,216 @@ func evalZPOPMIN(args []string, store *dstore.Store) *EvalResponse {
 		Error:  nil,
 	}
 }
+
+// evalSADD adds one or more members to a set
+// args must contain a key and one or more members to add the set
+// If the set does not exist, a new set is created and members are added to it
+// An error response is returned if the command is used on a key that contains a non-set value(eg: string)
+// Returns an integer which represents the number of members that were added to the set, not including
+// the members that were already present
+func evalSADD(args []string, store *dstore.Store) *EvalResponse {
+	if len(args) < 2 {
+		return &EvalResponse{
+			Result: nil,
+			Error:  diceerrors.ErrWrongArgumentCount("SADD"),
+		}
+	}
+	key := args[0]
+
+	// Get the set object from the store.
+	obj := store.Get(key)
+	lengthOfItems := len(args[1:])
+
+	var count = 0
+	if obj == nil {
+		var exDurationMs int64 = -1
+		var keepttl = false
+		// If the object does not exist, create a new set object.
+		value := make(map[string]struct{}, lengthOfItems)
+		// Create a new object.
+		obj = store.NewObj(value, exDurationMs, object.ObjTypeSet, object.ObjEncodingSetStr)
+		store.Put(key, obj, dstore.WithKeepTTL(keepttl))
+	}
+
+	if err := object.AssertType(obj.TypeEncoding, object.ObjTypeSet); err != nil {
+		return &EvalResponse{
+			Result: nil,
+			Error:  diceerrors.ErrWrongTypeOperation,
+		}
+	}
+
+	if err := object.AssertEncoding(obj.TypeEncoding, object.ObjEncodingSetStr); err != nil {
+		return &EvalResponse{
+			Result: nil,
+			Error:  diceerrors.ErrWrongTypeOperation,
+		}
+	}
+
+	// Get the set object.
+	set := obj.Value.(map[string]struct{})
+
+	for _, arg := range args[1:] {
+		if _, ok := set[arg]; !ok {
+			set[arg] = struct{}{}
+			count++
+		}
+	}
+
+	return &EvalResponse{
+		Result: count,
+		Error:  nil,
+	}
+}
+
+// evalSREM removes one or more members from a set
+// Members that are not member of this set are ignored
+// Returns the number of members that are removed from set
+// If set does not exist, 0 is returned
+// An error response is returned if the command is used on a key that contains a non-set value(eg: string)
+func evalSREM(args []string, store *dstore.Store) *EvalResponse {
+	if len(args) < 2 {
+		return &EvalResponse{
+			Result: nil,
+			Error:  diceerrors.ErrWrongArgumentCount("SREM"),
+		}
+	}
+	key := args[0]
+
+	// Get the set object from the store.
+	obj := store.Get(key)
+
+	var count = 0
+	if obj == nil {
+		return &EvalResponse{
+			Result: count,
```

Comment:
@AshwinKul28 - after applying this change, some of the http integration tests start to fail. do you recommend to modify the integration test or the change the response back to count(int)

Reply:
```
=== RUN   TestSetDataCmd/SADD_&_SCARD_with_non-existing_key
    set_data_cmd_test.go:304: 
        	Error Trace:	/home/.../dice/integration_tests/commands/http/set_data_cmd_test.go:304
        	Error:      	Not equal: 
        	            	expected: float64(0)
        	            	actual  : string("0")
        	Test:       	TestSetDataCmd/SADD_&_SCARD_with_non-existing_key
```

Reply:
HI @sahoss We can change the integration test response, because according to the latest HTTP response modification, it will return "0". wdyt @lucifercr07 ?

Reply:
@AshwinKul28 i examined the responses from other apis(eg: sadd). its a bit weird to return string in some and int/float in another. imo, for a numeric value, the client should not be expected to parse the string to an numeric type.

i feel its better to return the int here(instead of constant). later, we can figure/fix why the constant is getting converted to a string.

Reply:
Yes agreed, these are the responses from the HTTP requests. and we can work on them as a separate efforts.